### DISCUSSION TITLE
Create new MorphedRecord in form

### DISCUSSION THREAD

**Discussion by demobiel** [filamentphp/filament#8593] [Help]
Hi,

I have the following class with morphable relationship:
```
class EmailCampaignLine{

    public function receivable: MorhpTo{
         return $this->morphTo();
    }
}
```

And I have the following EmailCampaignLineResource:
```
class EmailCampaignLinesResource extends Resource
{
    public static function form(Form $form): Form
    {
        return $form
            ->schema([
                Group::make()
                    ->relationship('receivable')
                    ->schema([
                        Forms\Components\TextInput::make('firstname'),
                        Forms\Components\TextInput::make('lastname'),
                        Forms\Components\TextInput::make('email')
                    ])
                    ->saveRelationshipsBeforeChildrenUsing(function (\Filament\Forms\Components\Group $component) {
                        if ($component->getRelatedModel() === $component->getModel()) {
                            $component->getRelationship()->associate(EmailCampaignRecipient::make());
                        };
                    })
            ]);
    }
}
````

This is working just fine, however I am just wondering if there is a better way to achieve this, instead of using the `saveRelationshipsBeforeChildrenUsing`

Kind regards

Andries

**Comment by LorensoD** [comment]
I have the same question.

I currently have the following database structure:

```
Question
- id
- text
- order
- questionable_type
- questionable_id

QuestionX
- id
- question_id
- some_related_model_id

QuestionY
- id
- question_id
- some_json_column
- some_related_model_id
- some_related_model_id
```

So basically a Question model which has a singular polymorphic relationship with a variable question type model. Each question type stores different relations or other data only required by that specific question type.

**What I would like to achieve** is to be able to create these variable question type models (QuestionX or QuestionY) inline from the Question form.

Is there a way to easily create morphTo() relations inline in Filament? There should be some way to first specify what morphTo Model you want to create. And then the correct form for that morphTo Model should be rendered.
